kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultKms.java

```diff
@@ -107,8 +107,9 @@ public CompletionStage<DekPair<VaultEdek>> generateDekPair(@NonNull String kekRe
                 .POST(HttpRequest.BodyPublishers.noBody())
                 .build();
 
-        return vaultClient.sendAsync(request, statusHandler(kekRef, new JsonBodyHandler<VaultResponse<DataKeyData>>(new TypeReference<>() {
-        }), UnknownKeyException::new))
+        return vaultClient.sendAsync(request, statusHandler(new JsonBodyHandler<VaultResponse<DataKeyData>>(new TypeReference<>() {
```

Comment:
I think the changed code looks functionally correct, but it is not the easiest to read.  statusHandler's name's wrong - but I can't think of a good name for it - which is a code smell in itself.  Maybe there's a better way to arrange the code to make it more obvious.

I hadn't used the sendAsync API before - so maybe there's a better way.

Reply:
I've changed it to remove this unnecessary optimisation of discarding the body. Instead using the built-in byte[] BodyHandler and doing the JSON decode later.